### PR TITLE
Fix sim locale

### PR DESCRIPTION
**Pull request by guillaumejenkins**
Ended up having to duplicate some `lf` logic in the sim. Tried to keep it minimal.

pxtsim does not depend on pxtrunner or any other namespace - which means it needs its own `lf`

### CODE REVIEW

File: pxtsim/localization.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+namespace pxsim.localization {
+    let _localizeStrings: Map<string> = {};
```
Comment (microsoftsam) [inline pxtsim/localization.ts]: Can we leave a comment here describing where to find the duplicated function. Just so we always update one and the other hand in hand

File: pxtsim/localization.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+// Localization functions. Please port any modifications over to pxtlib/util.ts
+namespace pxsim.localization {
+    let _localizeStrings: Map<string> = {};
+
+    export function _localize(s: string) {
+        return _localizeStrings[s] || s;
+    }
+
+    export function setLocalizedStrings(strs: Map<string>) {
+        _localizeStrings = strs;
```
Comment (pelikhan) [inline pxtsim/localization.ts]: strs may be undefined -> ``strs || {}``

File: pxtsim/localization.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    export function jsStringQuote(s: string) {
+        return s.replace(/[^\w .!?\-$]/g,
+            (c) => {
+                let h = c.charCodeAt(0).toString(16);
+                return "\\u" + "0000".substr(0, 4 - h.length) + h;
+            });
+    }
+
+    export function fmt_va(f: string, args: any[]): string {
```
Comment (pelikhan) [inline pxtsim/localization.ts]: I don't think we need to support all that stuff for the sim. Maybe we just simplify it to just swapping the strings.

File: pxtsim/localization.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    export function _localize(s: string) {
+        return _localizeStrings[s] || s;
+    }
+
+    export function setLocalizedStrings(strs: Map<string>) {
+        _localizeStrings = strs;
+    }
+
+    export function htmlEscape(_input: string) {
```
Comment (pelikhan) [inline pxtsim/localization.ts]: don't export

File: pxtsim/localization.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            } else if (spec == ":url") {
+                r = encodeURI(r).replace(/'/g, "%27").replace(/"/g, "%22");
+            } else if (spec == ":%") {
+                r = (v * 100).toFixed(1).toString() + '%';
+            }
+            return r;
+        });
+    }
+
+    let sForPlural = true;
```
Comment (pelikhan) [inline pxtsim/localization.ts]: always true

**Pull request merged**